File: sdks/java/src/test/java/com/konfigthis/client/GettingStartedTest.java
package com.konfigthis.client;

import com.konfigthis.client.model.Account;
import com.konfigthis.client.model.AccountHoldings;
import com.konfigthis.client.model.Brokerage;
import com.konfigthis.client.model.DeleteUserResponse;
import com.konfigthis.client.model.OptionLeg;
import com.konfigthis.client.model.PortfolioGroup;
import com.konfigthis.client.model.SnapTradeRegisterUserRequestBody;
import com.konfigthis.client.model.Status;
import com.konfigthis.client.model.StrategyQuotes;
import com.konfigthis.client.model.TargetAsset;
import com.konfigthis.client.model.UniversalSymbol;
import com.konfigthis.client.model.UserIDandSecret;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.UUID;

import static org.junit.Assert.assertNotNull;

public class GettingStartedTest {

        @Disabled
        @Test
        public void setPortfolioTargets() throws ApiException {
                Configuration configuration = new Configuration();
                configuration.clientId = System.getenv("SNAPTRADE_CLIENT_ID");
                configuration.consumerKey = System.getenv("SNAPTRADE_CONSUMER_KEY");
                Snaptrade snaptrade = new Snaptrade(configuration);
                UUID userId = UUID.randomUUID();
                UserIDandSecret userIDandSecret = snaptrade.authentication.registerSnapTradeUser()
                                .userId(userId.toString()).execute();
                List<TargetAsset> targetAsset = new ArrayList<>();
                targetAsset.add(new TargetAsset().symbol(new UniversalSymbol().symbol("AAPL")).percent(90));
                snaptrade.authentication.deleteSnapTradeUser(userIDandSecret.getUserId()).execute();
        }

        @Test
        public void gettingStartedTest() throws ApiException {
                // 1) Initialize default client with clientID and consumerKey
                Configuration configuration = new Configuration();
                configuration.clientId = System.getenv("SNAPTRADE_CLIENT_ID");
                configuration.consumerKey = System.getenv("SNAPTRADE_CONSUMER_KEY");
                Snaptrade snaptrade = new Snaptrade(configuration);

                // 2) Check that the client is able to make a request to the API server
                Status status = snaptrade.apiStatus.check().execute();
                System.out.printf("SnapTrade is online: %s\n", status.getOnline());

                // 3) Create a new user on SnapTrade
                // The userId should be provided by you and refer to permanent value such as a
                // database row ID
                UUID userId = UUID.randomUUID();
                UserIDandSecret userIDandSecret = snaptrade.authentication.registerSnapTradeUser()
                                .userId(userId.toString()).execute();

                new SnapTradeRegisterUserRequestBody().userId(userId.toString());
                // Note: A user secret is only generated once. It's required to access resources
                // for certain endpoints
                System.out.printf("userID: %s, userSecret: %s\n", userIDandSecret.getUserId(),
                                userIDandSecret.getUserSecret());

                // 4) Get a redirect URI. Users will need this to connect their brokerage to the
                // SnapTrade server
                Map response = (Map) snaptrade.authentication
                                .loginSnapTradeUser(userIDandSecret.getUserId(),
                                                userIDandSecret.getUserSecret())
                                .execute();
                System.out.println(response.get("redirectURI"));

                // 5) Query holdings and available brokerages
                List<AccountHoldings> holdings = snaptrade.accountInformation
                                .getAllUserHoldings(userIDandSecret.getUserId(),
                                                userIDandSecret.getUserSecret())
                                .execute();
                System.out.println(holdings);
                List<Account> accounts = snaptrade.accountInformation.listUserAccounts(userIDandSecret.getUserId(),
                                userIDandSecret.getUserSecret()).execute();
                System.out.println(accounts);
                List<Brokerage> brokerages = snaptrade.referenceData.listAllBrokerages().execute();
                System.out.println(brokerages);

                // Test getOptionStrategy
                try {
                        UUID underlyingSymbolId = UUID.fromString("3ab7bf2d-beca-4de3-a215-f93d4a5a99b7");
                        List<OptionLeg> legs = new ArrayList<OptionLeg>();
                        String strategyType = "CUSTOM";
                        String option_id = "48947660";
                        Double quantity = 1.0;
                        OptionLeg leg = new OptionLeg().optionSymbolId(option_id)
                                        .action(OptionLeg.ActionEnum.BUY_TO_OPEN)
                                        .quantity(quantity);
                        legs.add(leg);
                        StrategyQuotes result = snaptrade.options
                                        .getOptionStrategy(underlyingSymbolId, legs, strategyType,
                                                        userIDandSecret.getUserId(),
                                                        userIDandSecret.getUserSecret(),
                                                        underlyingSymbolId)
                                        .execute();
                        System.out.println(result);
                } catch (ApiException e) {
                        System.out.println(e);
                }

                // 6) Deleting a user
                DeleteUserResponse deleteUserResponse = snaptrade.authentication
                                .deleteSnapTradeUser(userIDandSecret.getUserId()).execute();
                System.out.println(deleteUserResponse);
        }

}

File: sdks/php7/test/GettingStartedTest.php
<?php

namespace SnapTrade\Test\Api;


use PHPUnit\Framework\TestCase;


class GettingStartedTest extends TestCase
{

    /**
     * Setup before running any test cases
     */
    public static function setUpBeforeClass(): void
    {
    }

    /**
     * Setup before running each test case
     */
    public function setUp(): void
    {
    }

    /**
     * Clean up after running each test case
     */
    public function tearDown(): void
    {
    }

    /**
     * Clean up after running all test cases
     */
    public static function tearDownAfterClass(): void
    {
    }

    /**
     * Test case for getting started flow
     *
     */
    public function testGettingStarted()
    {
        // 1) Initialize a client with your clientID and consumerKey.
        $snaptrade = new \SnapTrade\Client(
            getenv("SNAPTRADE_CONSUMER_KEY"),
            getenv("SNAPTRADE_CLIENT_ID")
        );

        // 2) Check that the client is able to make a request to the API server.
        $response = $snaptrade->apiStatus->check();
        print_r($response);

        // 3) Create a new user on SnapTrade
        $userId = (string)time();
        $registerResponse = $snaptrade->authentication->registerSnapTradeUser($userId);
        $userSecret = $registerResponse->getUserSecret();

        // 4) Get a redirect URI. Users will need this to connect
        // their brokerage to the SnapTrade server.
        $redirectUri = $snaptrade->authentication->loginSnapTradeUser($userId, $userSecret);
        print_r($redirectUri);

        // 5) Delete the user
        $deletedResponse = $snaptrade->authentication->deleteSnapTradeUser($userId);
        print_r($deletedResponse);
    }

    public function testSslConfig()
    {
        $snaptrade = new \SnapTrade\Client(
            getenv("SNAPTRADE_CLIENT_ID"),
            getenv("SNAPTRADE_CONSUMER_KEY"),
            null,
            null,
            false,
            "https://expired.badssl.com/"
        );
        $result = $snaptrade->apiStatus->check();
        print_r($result);
    }

    public function testGetUserAccountQuotes()
    {
        $this->markTestSkipped("Getting 404 Not Found");
        $snaptrade = new \SnapTrade\Client(
            getenv("SNAPTRADE_CONSUMER_KEY"),
            getenv("SNAPTRADE_CLIENT_ID")
        );
        $user_id = getenv("SNAPTRADE_TEST_USER_ID");
        $user_secret = getenv("SNAPTRADE_TEST_USER_SECRET");
        $holdings = $snaptrade->accountInformation->getAllUserHoldings($user_id, $user_secret);
        $account_id = $holdings[0]->getAccount()->getId();
        $result = $snaptrade->trading->getUserAccountQuotes(
            $user_id,
            $user_secret,
            $account_id,
            "AAPL",
            true
        );
        print_r($result);
        $this->assertTrue($result != null, "Response must not be null");
    }

    public function testGetUserAccountBalance()
    {
        $snaptrade = new \SnapTrade\Client(
            getenv("SNAPTRADE_CONSUMER_KEY"),
            getenv("SNAPTRADE_CLIENT_ID")
        );
        $userId = getenv("SNAPTRADE_TEST_USER_ID");
        $userSecret = getenv("SNAPTRADE_TEST_USER_SECRET");
        $accounts = $snaptrade->accountInformation->listUserAccounts(
            $userId,
            $userSecret
        );
        $response = $snaptrade->accountInformation->getUserAccountBalance($userId, $userSecret, $accounts[0]->getId());
        print_r($response);
    }

    /**
     * Test case for setting portfolio targets
     *
     */
    public function testSetPortfolioTargets()
    {
        $this->markTestSkipped("Can't use listAssetClasses to get asset id");

        // 1) Initialize a client with your clientID and consumerKey.
        $snaptrade = new \SnapTrade\Client(
            getenv("SNAPTRADE_CLIENT_ID"),
            getenv("SNAPTRADE_CONSUMER_KEY")
        );

        // 2) Check that the client is able to make a request to the API server.
        $response = $snaptrade->apiStatus->check();
        print_r($response);

        // 3) Create a new user on SnapTrade
        $userId = (string)time();
        $registerResponse = $snaptrade->authentication->registerSnapTradeUser($userId);
        $userSecret = $registerResponse->getUserSecret();

        // 4) Delete the user
        $deletedResponse = $snaptrade->authentication->deleteSnapTradeUser($userId);
        print_r($deletedResponse);
    }
}
